src/facilities-context/transforms.tsx

```diff
@@ -0,0 +1,210 @@
+import { scaleThreshold } from "d3";
+import { getTime, isSameDay } from "date-fns";
+import { last, omit, orderBy } from "lodash";
+import { Optional } from "utility-types";
+
+import {
+  hasCases,
+  hasPopulation,
+  totalIncarceratedPopulation,
+} from "../impact-dashboard/EpidemicModelContext";
+import {
+  Facility,
+  ModelInputs,
+  ReferenceFacility,
+  SimpleTimeseries,
+} from "../page-multi-facility/types";
+import { validateMergedModelVersions } from "./validators";
+
+type MergeProps = {
+  userFacility: Facility;
+  referenceFacility?: ReferenceFacility;
+};
+
+type MergedHistoryInputs = {
+  userVersions: ModelInputs[];
+  referenceFacility: ReferenceFacility;
+};
+
+function getDateThresholdFunc(records: SimpleTimeseries[]) {
+  // the D3 scale function needs these as numbers so we'll use timestamps
+  const dateThresholds = records.map(({ date }) => getTime(date));
+  const outputValues = records.map(({ value }) => value);
+  // the range must have one extra value to capture numbers below the first threshold;
+  // in this case the oldest value we have should be used for any dates that precede it
+  outputValues.unshift(outputValues[0]);
+
+  const thresholdFunc = scaleThreshold()
```

Comment:
Ok, great!  This is pretty cool because even if we don't get great capacity or population data from the BiqQuery import, we should still be ok as long as the user fills in at least one value for each of these fields per facility